pkg/discovery/oci_dbbacked.go

```diff
@@ -133,8 +139,48 @@ func (od *DBBackedOCIDiscovery) listGroupsFromInventory() ([]*plugininventory.Pl
 // inventory of this discovery and stores it in the cache directory.
 func (od *DBBackedOCIDiscovery) fetchInventoryImage() error {
 	// TODO(khouzam): Improve by checking if we really need to download again or if we can use the cache
+
+	// Get the custom public key path and prepare cosign verifier, if empty, cosign verifier would use embedded public key for verification
+	customPublicKeyPath := os.Getenv(constants.PublicKeyPathForPluginDiscoveryImageSignature)
+	cosignVerifier := cosignhelper.NewCosignVerifier(customPublicKeyPath)
+	if err := od.verifyInventoryImageSignature(cosignVerifier); err != nil {
+		log.Warningf("Unable to verify the plugins discovery image signature: %v", err)
+		// TODO(pkalle): Update the message to convey user to check if they could use the latest public key after we get details of the well known location of the public key
+		errMsg := fmt.Sprintf("Fatal, plugins discovery image signature verification failed. The `tanzu` CLI can not ensure the integrity of the plugins to be installed. To ignore this validation please append %q to the comma-separated list in the environment variable %q.  This is NOT RECOMMENDED and could put your environment at risk!",
+			od.image, constants.PluginDiscoveryImageSignatureVerificationSkipList)
+		log.Fatal(nil, errMsg)
+	}
+
 	if err := carvelhelpers.DownloadImageAndSaveFilesToDir(od.image, od.pluginDataDir); err != nil {
 		return errors.Wrapf(err, "failed to download OCI image from discovery '%s'", od.Name())
 	}
 	return nil
 }
+
+func (od *DBBackedOCIDiscovery) verifyInventoryImageSignature(verifier cosignhelper.Cosignhelper) error {
+	signatureVerificationSkipList := getPluginDiscoveryImagesSkippedForSignatureVerification()
+	err := verifier.Verify(context.Background(), []string{od.image})
+	if err != nil {
+		if _, exists := signatureVerificationSkipList[strings.TrimSpace(od.image)]; exists {
+			// log warning message iff user had not chosen to skip warning message for signature verification
```

Comment:
```suggestion
			// log warning message if user had not chosen to skip warning message for signature verification
```

Reply:
I think Iff(if and only if) is still good here.